rllib/catalog/configs/encoder.py

```diff
@@ -0,0 +1,82 @@
+import abc
+from dataclasses import dataclass
+from typing import TYPE_CHECKING, Tuple
+from ray.rllib.catalog.torch.encoders.vector import TorchVectorEncoder
+from ray.rllib.models.specs.specs_dict import ModelSpec
+
+if TYPE_CHECKING:
+    from ray.rllib.catalog.torch.encoders.vector import Encoder
+
+
+@dataclass
+class EncoderConfig:
+    """The base config for encoder models. Each config should define a `build` method
+    that builds a model from the config.
+
+    All parameters known before runtime (e.g. framework, activation, num layers, etc.)
+    should be defined as attributes.
+
+    Parameters unknown before runtime (e.g. the output size of the module providing
+    input for this model) should be passed as arguments to `build`. This should be
+    as few params as possible.
+
+    `build` should return an instance of the encoder associated with the config.
+
+    Attributes:
+        framework: The tensor framework to construct a model for.
+            This can be 'torch', 'tf2', or 'jax'.
+    """
+
+    framework: str = "torch"
```

Comment:
I'm thinking about changing this property name also here to "framework_str" as we currently use `framework` as a method of AlgorithmConfig.

Currently:
AlgorithmConfig.framework() -> method to set the framework (and framework-related settings)
AlgorithmConfig.framework_str -> the str property, e.g. "torch"